bindings/tflite/java/org/tensorflow/lite/Interpreter.java

```diff
@@ -0,0 +1,436 @@
+    }
+
+    for (Map.Entry<Integer, Buffer> output : outputs.entrySet()) {
+      getOutputTensor(output.getKey()).copyToBuffer(output.getValue());
+    }
+  }
+
+  /**
+   * [Placeholder] for running model inference based on SignatureDef provided through @code
+   * methodName. This method is unsupported.
+   *
+   * @param inputs A Map of inputs from input name in the signatureDef to an input object.
+   * @param outputs a map mapping from output name in SignatureDef to output data.
+   * @param methodName The exported method name identifying the SignatureDef.
+   * @throws UnsupportedOperationException since IREE doesn't support tensor signatures.
+   */
+  public void runSignature(
+      @NonNull Map<String, Object> inputs,
+      @NonNull Map<String, Object> outputs,
+      String methodName) {
+    throw new UnsupportedOperationException("IREE does not support tensor signatures.");
+  }
+
+  /* Same as {@link Interpreter#runSignature(Object, Object, String)} but doesn't require
+   * passing a methodName, assuming the model has one SignatureDef. If the model has more than
+   * one SignatureDef it will throw an exception.
+   *
+   * * <p>WARNING: This is an experimental API and subject to change.
+   * */
+  public void runSignature(
+      @NonNull Map<String, Object> inputs, @NonNull Map<String, Object> outputs) {
+    throw new UnsupportedOperationException("IREE does not support tensor signatures.");
+  }
+
+  /**
+   * Explicitly updates allocations for all tensors, if necessary.
+   *
+   * <p>This will propagate shapes and memory allocations for all dependent tensors using the input
+   * tensor shape(s) as given.
+   *
+   * <p>Note: This call is *purely optional*. Tensor allocation will occur automatically during
+   * execution if any input tensors have been resized. This call is most useful in determining the
+   * shapes for any output tensors before executing the graph, e.g.,
+   *
+   * <pre>{@code
+   * interpreter.resizeInput(0, new int[]{1, 4, 4, 3}));
+   * interpreter.allocateTensors();
+   * FloatBuffer input = FloatBuffer.allocate(interpreter.getInputTensor(0),numElements());
+   * // Populate inputs...
+   * FloatBuffer output = FloatBuffer.allocate(interpreter.getOutputTensor(0).numElements());
+   * interpreter.run(input, output)
+   * // Process outputs...
+   * }</pre>
+   *
+   * @throws IllegalStateException if the graph's tensors could not be successfully allocated.
+   */
+  public void allocateTensors() {
+    if (nativeAllocateTensors() != 0) {
+      throw new IllegalStateException("Failed to allocate Tensors.");
+    }
+    tensorsAllocated = true;
+  }
+
+  /**
+   * Resizes idx-th input of the native model to the given dims.
+   *
+   * @throws IllegalArgumentException if {@code idx} is negative or is not smaller than the number
+   *     of model inputs; or if error occurs when resizing the idx-th input.
+   */
+  public void resizeInput(int inputIndex, @NonNull int[] dims) {
```

Comment:
match docs to code (`idx` -> `inputIndex`)

```suggestion
  /**
   * Resizes inputIndex-th input of the native model to the given dims.
   *
   * @throws IllegalArgumentException if {@code inputIndex} is negative or is not smaller than the number
   *     of model inputs; or if error occurs when resizing the inputIndex-th input.
   */
  public void resizeInput(int inputIndex, @NonNull int[] dims) {
```